app/models/provider.rb

```diff
@@ -63,4 +63,36 @@ def refresh_ems(opts = {})
     end
     managers.flat_map { |manager| EmsRefresh.queue_refresh(manager, nil, opts) }
   end
+
+  def self.destroy_queue(ids)
+    find(ids).each(&:destroy_queue)
+  end
+
+  def destroy_queue
+    msg = "Destroying #{self.class.name} with id: #{id}"
+
+    _log.info(msg)
+    task = MiqTask.create(
+      :name    => "Destroying #{self.class.name} with id: #{id}",
+      :state   => MiqTask::STATE_QUEUED,
+      :status  => MiqTask::STATUS_OK,
+      :message => msg,
+    )
+    self.class._queue_task('destroy', id, task.id)
+    task.id
+  end
+
+  def destroy(task_id = nil)
+    _log.info("To destroy managers of provider: #{self.class.name} with id: #{id}")
+    managers.each(&:destroy)
```

Comment:
@Fryguy @blomquisg @bdunne need some help here. This `manager` `destroy` is not triggering the destroy of managers' associated resources e.g. 'configured_system' etc.

That is failing the corresponding [Tower PR](https://github.com/ManageIQ/manageiq-providers-ansible_tower/pull/49).  If I keep the `:dependent => destroy` clause, the spec would pass and the `configued_system` resources are being destroyed in the `super().tap` call.

Not sure if this is to do with that: the `:dependent => destroy` between `ems` and `configured_system` is defined in the subclasses in `AutomationManager` namespace (i.e. [here](https://github.com/jameswnl/manageiq/blob/ed755cf192818b8b7bb3fe09b3c2d732dc89eb42/app/models/manageiq/providers/automation_manager.rb#L11-L11))

Reply:
@jameswnl Do you have a test that fails that you can share with us?

Reply:
@bdunne yes, [here](https://travis-ci.org/ManageIQ/manageiq-providers-ansible_tower/jobs/325669516#L2226)
However, my new update the that Tower PR now is passing (in my local)